### DISCUSSION TITLE
Import Classification Store via Data Importer

### DISCUSSION THREAD

**Discussion by smitterer** [pimcore/pimcore#13708] [Q&A / Support]
Hi everyone,

I'm new to pimcore and am currently going through the academy to learn it.

What I wasn't able to learn while going through the Junior Dev certification is, how to import into a classification store via the Data Importer. I also didn't see something like this in the other certification courses yet.

Basically what I want to achieve is exactly what's presented under https://pimcore.com/en/resources/blog/etim-classification-with-pimcore-s-classification-store-at-jager-direkt-gmbh-co.kg_a127883.

Is this possible with the Data Importer and with having the data in an xls or csv? This is the source data file: https://www.etim-international.com/wp-content/uploads/2020/11/ETIM-8.0-ALL-SECTORS-EXCEL-METRIC-EI-2020-11-02.zip

I can see Data Target "Classifcation Store" under the mappings, but wasn't able to import it.

Thanks

**Comment by BlackbitDevs** [comment]
For creating the ETIM classification store, please have a look at https://github.com/BlackbitDigitalCommerce/pimcore-data-director-bmecat#importing-etim-classes--groups--features-to-pimcore-classification-store - there is a command which creates the classification store(s) for you and creates all the fields which ETIM defines.

For importing data into classification store, I am not sure if Data Importer can do this but https://pimcore.com/en/developers/marketplace/blackbit_digital_commerce/pimcore-data-director_e103850 can. One little excerpt from https://pimcore.blackbit.de/Blackbit/1.pimcore/Handb%C3%BCcher/Pimcore-Data-Director-Manual.pdf:
> #### Object bricks / Classification Store
> 
> All the single fields of the allowed object bricks / classification store groups get displayed in the attribute mapping panel and can be mapped individually. You can treat them like normal class fields so all the things listed here for other field types are valid also for object brick / classification store fields. As soon as at least for one of the fields of an object brick / classification store group data gets imported the brick / group will get added to the current object.
> 
> Alternatively you can directly map your data to the object brick / classification store container field by returning an associative array whose keys are the brick / classification store group names:
> 
> 
```php
return [
    'brickName' => [
        'field1' => $params['rawItemData']['field_1']['value'],
        'field2' => $params['rawItemData']['field_2']['value'],
        'name#de' => $params['rawItemData']['name de']['value'], // name is a localized field
        'name#en' => $params['rawItemData']['name en']['value'],
    ],
];
```
> 
> Convenience features like resolving data query selectors are applied to all the provided fields, e.g.
> 
```php
return [
    'technicalAttributes' => [
        'cableLength' => 'AttributeValue:externalId:'.$params['rawItemData']['id']['value'].':value'
    ],
];
```
> 
> This will search for an object of the class `AttributeValue` whose `externalId` equals the value from the raw data field `id` and import this to the field `cableLength` of the object brick `technicalAttributes`.
> 
> Or if the field name is part of the import data you can assign it dynamically:
> 
```php
return [
    'technicalAttributes' => [
        $params['rawItemData']['attributeName']['value'] => $params['rawItemData']['attributeValue']['value']
    ],
];
```

**Reply by smitterer** [reply]
Hi, thanks for your reply. I already saw your plugins and they seem to be very powerful. The thing is, I'm new to pimcore and am not even 100% sure if we are going to use it in our company and for that, the initial investment into the modules is not the cheapest to be honest while I'm not saying it's not worth it!

**Reply by BlackbitDevs** [reply]
I understand. If we can support you in making the decision (comparing Pimcore with other PIM systems), please contact info@blackbit.de
